### PR TITLE
Java cheatsheet: Removed unusual hidden character, Corrected small grammatical mistake

### PR DESCRIPTION
**Pull request by mayank7**
Removed unusual hidden character from Java cheat sheet which are not directly visible here but can be seen in text editors like Sublime and also in "git gui".
Corrected small grammatical mistake also.

Java Cheat sheet IA page : https://duck.co/ia/view/java_cheat_sheet

### CODE REVIEW

**Comment by daxtheduck** [comment]
@mayank7 Hey!
        
Thanks for taking the time to contribute! We really appreciate it.

We work closely with every contributor to make Instant Answers the best they can be, so we appreciate your patience as we look over your code. From here, the process usually goes like this:

1. Pull Request is reviewed by the DuckDuckGo staff and community
2. Staff and community will leave feedback with any necessary updates to the function or design.
3. Once you've made any necessary corrections, then your Instant Answer will be merged and deployed live on DuckDuckGo!

If you have any questions along the way, feel free to ask them here. Our staff and community are also available on Slack to answer any questions you may have. If you'd like to join us there please send an email to <quackslack@duckduckgo.com>.

>More Info: https://duck.co/duckduckhack/submission_and_review

Thanks!

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

File: share/goodie/cheat_sheets/json/java.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
     "id": "java_cheat_sheet",
     "name": "Java",
-    "description": "A set of java Syntax, functions, and expressions",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/java.json]: This comma is actually OK. It's an "oxford comma"

File: share/goodie/cheat_sheets/json/java.json [hunk 2]
```diff
@@ -52,153 +52,139 @@
...
             {
                 "val" : "If Statement",
-                "key" : "if(expre­ssion)&nbsp;\\{­<br>&nbsp;&nbsp;&nbsp;&nbsp;st­ate­ments<br>\\}&nbsp;else&nbsp;\\{<br>&nbsp;&nbsp;&nbsp;&nbsp;­st­ate­ments<br>\\}"
-                
+                "key" : "if(expression)&nbsp;\\{<br>&nbsp;&nbsp;&nbsp;&nbsp;statements<br>\\}&nbsp;else&nbsp;\\{<br>&nbsp;&nbsp;&nbsp;&nbsp;statements<br>\\}"
             },
             {
-                
                 "val" : "While Loop",
-                "key" : "while(expre­ssion)&nbsp;\\{<br>­&nbsp;&nbsp;&nbsp;&nbsp;st­ate­ments<br>\\}"
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/java.json]: We should really remove the `<br>` and `&nbsp;` -- these shouldn't contain any HTML or HTML entities.

We are going to add support for `\n` and `\t` soon

Comment (mayank7) [inline share/goodie/cheat_sheets/json/java.json]: Okay

Comment (mayank7) [inline share/goodie/cheat_sheets/json/java.json]: https://duck.co/ia/view/java_cheat_sheet

**Comment by mayank7** [comment]
@moollaza  Removing &nbsp; and <br> is failing in continuous-integration/travis-ci/pr.

**Comment by mayank7** [comment]
@moollaza  Removing the <br> and &nbsp; is failing in continuous-integration/travis-ci/pr

File: share/goodie/cheat_sheets/json/java.json (hunk 2 above)
Comment (mayank7) [inline share/goodie/cheat_sheets/json/java.json]: @moollaza Removed HTML tags as suggested by you.

**Comment by mintsoft** [comment]
@moollaza  this looks a bit pants without the whitespace; especially the try/catch example

**Comment by mayank7** [comment]
@moollaza @mintsoft Any change I need to do ?

**Comment by moollaza** [comment]
>@moollaza this looks a bit pants without the whitespace; especially the try/catch example

@mayank7 sorry I meant that we should remove the `&nbsp` but keep in spaces ` ` (like we do for all other cheat sheets) so that this is at least legible.

Once we have support for `\n`, `\t` we can reformat code blocks -- we will likely want to rope in the design team because we didn't really plan for multi-line blocks of formatted code when we made the cheat sheet designs

**Comment by moollaza** [comment]
@mayank7 support for `\n` and `\t` has now been added. Please update this IA to use those so everything looks good :+1:

**Comment by mayank7** [comment]
@moollaza Sure

**Comment by mayank7** [comment]
@moollaza I have made the changes for br tag and &nbsp, is it fine now ?

**Comment by moollaza** [comment]
@mayank7 it's nearly there. Some parts still look a little odd though. I think we're using `\t` in places where we just need 1 or 2 spaces:

java_cheat_sheet_at_duckduckgo

**Comment by GuiltyDolphin** [comment]
@mayank7 Move into #2620 if still applicable.

Remember to link the IA Page and @mention the maintainer in the PR description.

**Pull request closed without merging**